pkg/kubelet/kubelet_pods.go

```diff
@@ -2095,18 +2095,14 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 				}
 			}
 		}
-		container := kubecontainer.GetContainerSpec(pod, cName)
 
 		// Always set the status to the latest allocated resources, even if it differs from the
 		// allocation used by the current sync loop.
 		alloc, found := kl.statusManager.GetContainerResourceAllocation(string(pod.UID), cName)
-		if found {
-			status.AllocatedResources = alloc.Requests
-		} else if !(container.Resources.Requests == nil && container.Resources.Limits == nil) {
-			// This case is expected for ephemeral containers.
-			if oldStatusFound {
-				status.AllocatedResources = oldStatus.AllocatedResources
-			}
+		if !found {
+			// This case is expected for non-resizeable containers.
```

Comment:
What are the non-resizable containers? Ephemeral containers? Can we list a few examples here?

Reply:
ephemeral containers & non-restartable (non-sidecar) init containers. Added a comment.